eng/Version.Details.xml

```diff
@@ -22,5 +22,9 @@
       <Uri>https://github.com/dotnet/aspnetcore</Uri>
       <Sha>91d17f6c2dfcd4bd296a447faffc0893cdad4e9e</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.Windows.Compatibility" Version="3.1.0">
```

Comment:
| Before                                                                                                    | After                                                                                                     |
|-----------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------|
| 12M     dotnet-counters 3.1M    dotnet-dump 11M     dotnet-gcdump 1.3M    dotnet-sos 11M     dotnet-trace | 26M     dotnet-counters 18M     dotnet-dump 25M     dotnet-gcdump 1.3M    dotnet-sos 25M     dotnet-trace |

The size is increased by at least 15 MB for each tool. That is significant.
Those dependencies are needed for Windows. Linux needs no dependencies.

Reply:
That's too big. Can we use a different API? As an example this can be done without adding a package https://github.com/projectkudu/kudu/blob/787c893a9336beb498252bb2f90a06a95763f9e9/Kudu.Core/Infrastructure/ProcessExtensions.cs#L562-L616

Reply:
I agree with David, that is too high a size cost.  Try with the code that David linked and let's see if we can't keep the size down.

Reply:
Changes suggested by davidfowl were implemented and the size of the tool is back to original size.